core/src/saros/versioning/VersionManager.java

```diff
@@ -27,188 +32,144 @@
+            .getContact(rqJID.getRAW())
+            .map(remoteInfo::get)
+            .map(ClientInfo::getVersion)
+            .orElse(null);
+    if (remoteVersion == null) {
+      log.warn("remote version not found for: " + rqJID);
+      return new VersionCompatibilityResult(Compatibility.UNKNOWN, localVersion, Version.INVALID);
     }
 
+    Compatibility compatibility = determineCompatibility(localVersion, remoteVersion);
     return new VersionCompatibilityResult(compatibility, localVersion, remoteVersion);
   }
 
-  private VersionExchangeExtension queryRemoteVersionDetails(final JID rqJID, final long timeout) {
-    assert rqJID.isResourceQualifiedJID();
-
-    final String exchangeID = String.valueOf(ID_GENERATOR.nextInt());
-    VersionExchangeExtension versionExchangeRequest = new VersionExchangeExtension();
-
-    versionExchangeRequest.set(VERSION_KEY, localVersion.toString());
-    versionExchangeRequest.set(ID_KEY, String.valueOf(exchangeID));
-
-    IQ request = VersionExchangeExtension.PROVIDER.createIQ(versionExchangeRequest);
+  /**
+   * Set a Information that should be broadcasted to other Contacts.
+   *
+   * @param key
+   * @param value if null key is removed from local info
+   */
+  public void setLocalInfo(String key, String value) {
+    Objects.requireNonNull(key, "key is null");
 
-    request.setType(IQ.Type.GET);
-    request.setTo(rqJID.toString());
+    if (value == null) localInfo.remove(key);
+    else localInfo.put(key, value);
 
-    PacketCollector collector =
-        receiver.createCollector(
-            new AndFilter(
-                VersionExchangeExtension.PROVIDER.getIQFilter(),
-                new PacketFilter() {
-                  @Override
-                  public boolean accept(Packet packet) {
+    if (remoteInfo.isEmpty()) return;
+    // send new info to all known online contacts
+    Enumeration<XMPPContact> contacts = remoteInfo.keys();
+    while (contacts.hasMoreElements()) sendVersionInfo(contacts.nextElement());
+  }
 
-                    VersionExchangeExtension versionExchange =
-                        VersionExchangeExtension.PROVIDER.getPayload(packet);
+  /**
+   * If available get Information received by other Contacts.
+   *
+   * @param contact
+   * @param key
+   * @return Optional the String associated with this contact and key
+   */
+  public Optional<String> getRemoteInfo(XMPPContact contact, String key) {
+    ClientInfo clientInfo = remoteInfo.get(contact);
+    if (clientInfo == null) return Optional.empty();
 
-                    if (versionExchange == null) return false;
+    return Optional.ofNullable(clientInfo.getInfo(key));
+  }
 
-                    return rqJID.toString().equals(packet.getFrom())
-                        && ((IQ) packet).getType() == IQ.Type.RESULT
-                        && exchangeID.equals(versionExchange.get(ID_KEY));
-                  }
-                }));
+  private void sendVersionInfo(XMPPContact contact) {
```

Comment:
Small info that non Saros contacts are ignored would be nice.

Reply:
done